omnigibson/maps/traversable_map.py

```diff
@@ -226,30 +175,36 @@ def get_shortest_path(self, floor, source_world, target_world, entire_path=False
             source_world (2-array): (x,y) 2D source location in world reference frame (metric)
             target_world (2-array): (x,y) 2D target location in world reference frame (metric)
             entire_path (bool): whether to return the entire path
+            robot (None or BaseRobot): if given, erode the traversability map to account for the robot's size
 
         Returns:
             2-tuple:
                 - (N, 2) array: array of path waypoints, where N is the number of generated waypoints
                 - float: geodesic distance of the path
         """
-        assert self.build_graph, "cannot get shortest path without building the graph"
         source_map = tuple(self.world_to_map(source_world))
         target_map = tuple(self.world_to_map(target_world))
 
-        g = self.floor_graph[floor]
-
-        if not g.has_node(target_map):
-            nodes = np.array(g.nodes)
-            closest_node = tuple(nodes[np.argmin(np.linalg.norm(nodes - target_map, axis=1))])
-            g.add_edge(closest_node, target_map, weight=T.l2_distance(closest_node, target_map))
+        # create a deep copy so that we don't erode the original map
+        trav_map = copy.deepcopy(self.floor_map[floor])
 
-        if not g.has_node(source_map):
-            nodes = np.array(g.nodes)
-            closest_node = tuple(nodes[np.argmin(np.linalg.norm(nodes - source_map, axis=1))])
-            g.add_edge(closest_node, source_map, weight=T.l2_distance(closest_node, source_map))
-
-        path_map = np.array(nx.astar_path(g, source_map, target_map, heuristic=T.l2_distance))
+        # erode the traversability map to account for the robot's size
+        if robot:
```

Comment:
same here -- might also be good to reduce duplicate code and put this block of functionality in another call, e.g.: `erode_trav_map(self, robot)`

Reply:
+1